chemprop/utils.py

```diff
@@ -528,3 +528,86 @@ def save_smiles_splits(data_path: str,
 
     with open(os.path.join(save_dir, 'split_indices.pckl'), 'wb') as f:
         pickle.dump(all_split_indices, f)
+
+
+def update_prediction_args(predict_args: PredictArgs,
+                           train_args: TrainArgs,
+                           missing_to_defaults: bool = True,
+                           for_molecule_fingerprint: bool = False) -> None:
+    """
+    Updates prediction arguments with training arguments loaded from a checkpoint file.
+    If an argument is present in both, the predicion argument will be used.
+
+    Also raises errors for situations where the prediction arguments and training arguments
+    are different but must match for proper function.
+
+    :param predict_args: The :class:`~chemprop.args.PredictArgs` object containing the arguments to use for making predictions.
+    :param train_args: The :class:`~chemprop.args.TrainArgs` object containing the arguments used to train the model previously.
+    :param missing_to_defaults: Whether to replace missing training arguments with the current defaults for :class: `~chemprop.args.TrainArgs`.
+        This is used for backwards compatibility.
+    :param for_molecule_fingerprint: Indicates whether this argument update is being used for molecule fingerprint generation,
+        in which case molecule features are not needed, even if used in the model training.
+    """
+    for key, value in vars(train_args).items():
+        if not hasattr(predict_args, key):
+            setattr(predict_args, key, value)
+
+    if missing_to_defaults:
+        # If a default argument would cause different behavior than occurred in legacy checkpoints before the argument existed,
+        # then that argument must be included in the `override_defaults` dictionary to force the legacy behavior.
+        override_defaults = dict((
+            ('bond_features_scaling',False),
+            ('no_bond_features_scaling',True),
+            ('atom_descriptors_scaling',False),
+            ('no_atom_descriptors_scaling',True),
+        ))
+        default_train_args=TrainArgs().parse_args(['--data_path',None,'--dataset_type',str(train_args.dataset_type)])
+        for key, value in vars(default_train_args).items():
+            if not hasattr(predict_args,key):
+                if key in override_defaults.keys():
+                    setattr(predict_args,key,override_defaults[key])
+                else:
+                    setattr(predict_args, key, value)
```

Comment:
This could be `setattr(predict_args, key, override_defaults.get(key, value))`.

Reply:
I think you misinterpreted this comment. The entire if/else could be replaced by providing `value` as the second argument to `get`. The second argument is the value to return if the key is not present in the dictionary.

Reply:
So I did. I saw the redundancy in the second argument with get but didn't understand that you were simplifying the whole thing instead of primarily substituting in the get function.